ios/EmbedReactNative.swift

```diff
@@ -248,6 +250,41 @@ class EmbedReactNative: NSObject {
     }
   }
 
+  func decode<T: Codable>(from dict: [String: Any?], to type: T.Type) -> T? {
+      func sanitize(dict: [String: Any?]) -> [String: Any] {
+          var result = [String: Any]()
+
+          for (key, value) in dict {
+              if let dictValue = value as? [String: Any?] {
+                  result[key] = sanitize(dict: dictValue)
+              } else if let nonNilValue = value {
+                  result[key] = nonNilValue
+              }
+          }
+
+          return result
+      }
+
+      let sanitizedDict = sanitize(dict: dict) // dictionary without nil values
```

Comment:
I don't fully understand what is going here, why do we need to sanitize the buyer object?

Reply:
The object might contain null values which I understand would cause `JSONSerialization.data` to fail to construct the json data https://developer.apple.com/documentation/foundation/jsonserialization/1413636-data

Reply:
That seems like a strange rule given `null` is a valid value for JSON - did you test sending a null value?